### DISCUSSION TITLE
data source "snowflake_roles" & "snowflake_grants"

### DISCUSSION THREAD

**Discussion by ob1jedi** [Snowflake-Labs/terraform-provider-snowflake#1141] [Ideas] [answered]
We are attempting to manage drift between Snowflake and Terraform, as well as, generate graphs that depict the relationships between all our resources (in and outside of Snowflake). We'd like to be able to list roles and grants.
Would appreciate it greatly, thanks.

**Comment by frosforever** [comment, marked as answer]
Sounds like this may be related? https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/740. Essentially, the current provider doesn't track existing roles / grants in resources or in data sources

**Reply by systematics78** [reply]
https://github.com/Snowflake-Labs/terraform-provider-snowflake/actions/runs/3228756660
UP :)
